### ISSUE TITLE
SDDM 0.16 isn't started automatically by systemd

### ISSUE THREAD

**Issue by grayk79** [sddm/sddm#905]
`systemctl enable sddm` does link successfully but on the next system restart nothing happens. The last thing seen is how the video driver is loaded successfully and that's all. Loading sddm from a tty with `systemctl start sddm` does work like it should. The logs are empty, no sign of an error. And `systemctl status` does list sddm as loaded for some reason. Only downgrading to sddm 0.15 helped to get the system booted up automatically like it always has had and should. For now I have blacklisted sddm from upgrading

OS: Arch Linux x64
Xorg: 1.19.4
Qt: 5.9.2
Plasma: 5.11
Video driver: AMDGPU

**Comment by farseerfc** [comment]
Same here, but only happens somethings.
When this happens, login through tty and restart sddm by `systemctl restart sddm` or `pkill sddm` will start sddm normally.

OS: Arch Linux x64
Xorg: xorg-server 1.19.4
Qt: qt5-base 5.9.2
Plasma: 5.11
Video driver: i915 with xorg modesetting

**Comment by homza** [comment]
Same here as well, tried every fix I could find from Googling around after hours I give up.
CTRL + ALT + F2 sudo systemctl restart sddm.service then it works fine, a pain though.
`Something` to try as temp fix is to make a systemd timer for SDDM, I didn't try it yet;
From root:
Create the file:  /usr/lib/systemd/system/sddm.timer

Contents are as follows:

[Unit]
Description=Start sddm service fixed time after boot

[Timer]
OnStartupSec=30
Unit=sddm.service

[Install]
WantedBy=multi-user.target

Then issue the following commands:
systemctl disable sddm.service
systemctl enable sddm.timer

To remove the workaround, simply enter:
systemctl disable sddm.timer
systemctl enable `sddm.service`

Other Thoughts; I wounder if someone will fix this, hum... 
OS: Arch Linux x64, with the all latest and the greatest updates except this one...

**Comment by cchurchill82** [comment]
This can be fixed by building your video module (i915, raedon, etc) module into the initramfs.
Link to Arch bug report: https://bugs.archlinux.org/task/55960

**Comment by homza** [comment]
Thankyou...
sudo systemctl disable sddm.timer
sudo systemctl enable sddm.service
sudo nano /etc/mkinitcpio.conf
added MODULES="radeon"
sudo mkinitcpio -p linux
sudo systemctl reboot
Hey look SDDM
My lsmod dump
 Module                  Size  Used by
ccm                    20480  6
fuse                   94208  3
cmac                   16384  1
rfcomm                 69632  16
bnep                   20480  2
mousedev               20480  0
hid_logitech_hidpp     32768  0
arc4                   16384  2
input_leds             16384  0
joydev                 20480  0
hid_logitech_dj        20480  0
iwlmvm                299008  0
nls_iso8859_1          16384  1
mac80211              688128  1 iwlmvm
nls_cp437              20480  1
snd_hda_codec_realtek    77824  1
led_class              16384  2 iwlmvm,input_leds
snd_hda_codec_generic    69632  1 snd_hda_codec_realtek
vfat                   20480  1
fat                    65536  1 vfat
snd_hda_codec_hdmi     49152  2
btusb                  40960  0
snd_hda_intel          36864  5
iwlwifi               217088  1 iwlmvm
btrtl                  16384  1 btusb
btbcm                  16384  1 btusb
mxm_wmi                16384  0
snd_hda_codec         106496  4 snd_hda_intel,snd_hda_codec_hdmi,snd_hda_codec_generic,snd_hda_codec_realtek
btintel                16384  1 btusb
evdev                  24576  10
edac_mce_amd           28672  0
mac_hid                16384  0
bluetooth             479232  41 btrtl,btintel,bnep,btbcm,rfcomm,btusb
snd_hda_core           65536  5 snd_hda_intel,snd_hda_codec,snd_hda_codec_hdmi,snd_hda_codec_generic,snd_hda_codec_realtek
cfg80211              532480  3 iwlmvm,iwlwifi,mac80211
snd_hwdep              20480  1 snd_hda_codec
kvm_amd              2174976  0
r8169                  77824  0
ecdh_generic           24576  1 bluetooth
snd_pcm                86016  4 snd_hda_intel,snd_hda_codec,snd_hda_core,snd_hda_codec_hdmi
snd_timer              28672  1 snd_pcm
kvm                   516096  1 kvm_amd
pcspkr                 16384  0
k10temp                16384  0
mii                    16384  1 r8169
snd                    73728  18 snd_hda_intel,snd_hwdep,snd_hda_codec,snd_timer,snd_hda_codec_hdmi,snd_hda_codec_generic,snd_hda_codec_realtek,snd_pcm
irqbypass              16384  1 kvm
soundcore              16384  1 snd
rfkill                 20480  9 bluetooth,cfg80211
sp5100_tco             16384  0
i2c_piix4              24576  0
tpm_infineon           20480  0
shpchp                 32768  0
wmi                    20480  1 mxm_wmi
tpm_tis                16384  0
tpm_tis_core           20480  1 tpm_tis
tpm                    49152  3 tpm_infineon,tpm_tis,tpm_tis_core
button                 16384  0
acpi_cpufreq           20480  0
sch_fq_codel           20480  6
sg                     36864  0
crypto_user            16384  0
ip_tables              24576  0
x_tables               32768  1 ip_tables
ext4                  557056  1
crc16                  16384  2 bluetooth,ext4
mbcache                16384  1 ext4
jbd2                   90112  1 ext4
fscrypto               24576  1 ext4
hid_generic            16384  0
usbhid                 45056  0
hid                   114688  4 hid_generic,usbhid,hid_logitech_dj,hid_logitech_hidpp
sd_mod                 49152  3
ata_generic            16384  0
ohci_pci               16384  0
pata_acpi              16384  0
firewire_ohci          40960  0
ahci                   36864  2
firewire_core          57344  1 firewire_ohci
crc_itu_t              16384  1 firewire_core
pata_atiixp            16384  0
libahci                28672  1 ahci
xhci_pci               16384  0
ehci_pci               16384  0
ehci_hcd               73728  1 ehci_pci
ohci_hcd               49152  1 ohci_pci
xhci_hcd              188416  1 xhci_pci
libata                208896  5 pata_atiixp,ahci,libahci,ata_generic,pata_acpi
usbcore               208896  8 usbhid,ehci_hcd,ohci_pci,xhci_pci,btusb,ohci_hcd,xhci_hcd,ehci_pci
scsi_mod              155648  3 sd_mod,libata,sg
usb_common             16384  1 usbcore
serio                  20480  0
radeon               1437696  33
i2c_algo_bit           16384  1 radeon
drm_kms_helper        131072  1 radeon
syscopyarea            16384  1 drm_kms_helper
sysfillrect            16384  1 drm_kms_helper
sysimgblt              16384  1 drm_kms_helper
fb_sys_fops            16384  1 drm_kms_helper
ttm                    81920  1 radeon
drm                   303104  19 radeon,ttm,drm_kms_helper
agpgart                36864  2 ttm,drm

**Comment by Dar13** [comment]
Is this the official solution or just a workaround?

**Comment by BobbyBabes** [comment]
```Bash
sudo nano /etc/mkinitcpio.conf
MODULES="amdgpu" # Was : MODULES="“.
sudo mkinitcpio -p linux
shutdown -h now
```
Then powering back on, did the trick for me.

**Comment by homza** [comment]
I think these salutations are jest workarounds.
I think they jest add a startup delay of a millisecond or two which masks the problem, because if you look at your loaded modules before rebuilding initramfs using [you@your-pc~]# lsmod command you'll see that radeon/amdgpu is loaded correctly as its part of -p which stands for presets a preset config for mkinitcpio building of initramfs andradeon/amdgpu is typically initialized after the initramfs stage by default it is possible, however, to enable radeon/amdgpu during the initramfs stage that is causing the extra needed boot process time needed to mask this problem.
Basically any mod or added service that slows down boot load time will add the extra needed boot process time needed to mask this problem.
Earlier versions work correctly without adding to MODULES=""  or a systemd sddm load timer addition.
Other things that work to mask it and not fix it add some auto nfs/samba mounts in fstab.
Good Luck to whoever attempts to figure this out, for now only workarounds, or move to another DM.

**Comment by Scimmia22** [comment]
This happens when SDDM tries to start before video is totally initialized. By adding your video module to early userspace, you initialize video earlier and avoid the problem, but it doesn't really fix the root issue. Not sure if the issue is here, in systemd, or in the kernel.

**Comment by DeX77** [comment]
Seems like there was a similar Issue some time ago: 

https://github.com/systemd/systemd/issues/2612

I tried adding systemd-vconsole-setup.service to After= but that didn't help

**Comment by farseerfc** [comment]
Unfortunately I cannot adding i915 to early userspace, because there is another unrelated issue that will cause the backlight adjusting module not loaded properly on my device.

**Comment by Musikolo** [comment]
Same issue happens to me on Lenovo T400. Reverting back to sddm-0.15 solves the problem. I haven't tried adding any module in /etc/mkinitcpio.conf though.

**Comment by angel6700** [comment]
Same problem here, with a laptop acer 1410. 
But fixed either reverting to sddm 0.15, or 
sddm 0.16 + editing mkinitcpio.conf and adding MODULES="i915"

**Comment by TheEdward162** [comment]
Same problem on Dell Inspiron 7000
Either reverting to version 0.15 or adding "i915" to MODULES fixes the problem

**Comment by majewsky** [comment]
The workaround works for me as well, but I'd still appreciate a solution that does not require that. I usually do not load the video driver in the initrd as the modesetting-induced flickering delays the display of the LUKS passphrase prompt by a few seconds.

**Comment by sgon00** [comment]
+1

OS: Manjaro x64
Xorg: xorg-server 1.19.5-1
Qt: qt5-base 5.9.2-1
Video driver: i915

**Comment by jonathonf** [comment]
`lightdm.service` has:

```
After=getty@tty1.service systemd-user-sessions.service plymouth-quit.service acpid.service
```
The only difference there is `acpid.service` - any use?

**Comment by antonio-rojas** [comment]
Given that this seems to be a regression in 0.16, if would be good if someone affected could bisect it

**Comment by BobbyBabes** [comment]
```
Linux htpc 4.13.11-1-ARCH #1 SMP PREEMPT Thu Nov 2 10:25:56 CET 2017 x86_64 GNU/Linux
sddm 0.16.0-1
xorg-server 1.19.5-1
qt5-base 5.9.2-1
mesa 17.2.4-1
xf86-video-amdgpu 1.4.0-1
```

I just gave it a shot on my htpc (old hardware; ignore the USB and amdgpu errors, they are always there).
I postpended `acpid.service` to the `After=` line. Same problem. Hangs at : see screenshot.

![sddm-0 16 0_hangs-boot-process](https://user-images.githubusercontent.com/22870070/32495537-53fe42fc-c3c6-11e7-996e-7273218fd43b.jpg)

**Comment by elvisangelaccio** [comment]
FWIW, I can reproduce it only on my Intel laptop, but not on my desktop with proprietary Nvidia drivers. Both systems run more or less the same Archlinux packages.

**Comment by BobbyBabes** [comment]
@elvisangelaccio Reproduce with `acpid.service` added or without ?

**Comment by elvisangelaccio** [comment]
@BobbyBabes I didn't add anything, so whatever the default is (without?).

**Comment by BobbyBabes** [comment]
The default is without on Arch.

**Comment by h4nn35** [comment]
I tried to bisect the problem and if my bisect worked then commit https://github.com/sddm/sddm/commit/2187cd880c62ad415ec360493fe3a0ead01bd772 is where the issue started.

**Comment by antonio-rojas** [comment]
Does it work if you add systemd-logind.service to the After= line in sddm.service?

**Comment by BobbyBabes** [comment]
@antonio-rojas Just tried. I postpended `systemd-logind.service` to the `After=` line in `/usr/lib/systemd/system/sddm.service`.
Alas, similar result as per my previous attempt with postpending `acpid.service`. This time with an awful flash though. My heart skipped a beat.

**Comment by Strit** [comment]
Running 
```
sudo systemd-sysusers sddm.conf
```
made sddm start for me.

**Comment by NuLogicSystems** [comment]
Yes, systemd-sysuser needs to be set to use that file.

**Comment by BobbyBabes** [comment]
It's there since October 8. So I didn't bother issuing the command. :
```
[bobbybabes@htpc 4.4.12 ~]# ls -lah /usr/lib/sysusers.d
total 240K
drwxr-xr-x   2 root root 4,0K nov  5 20:38 .
drwxr-xr-x 201 root root 192K nov  9 13:04 ..
-rw-r--r--   1 root root 1,1K okt 26 16:06 basic.conf
-rw-r--r--   1 root root   21 sep 15 20:47 ceph.conf
-rw-r--r--   1 root root  206 okt 30 16:24 dbus.conf
-rw-r--r--   1 root root   36 sep 27 20:59 mariadb.conf
-rw-r--r--   1 root root   11 okt  3 06:17 qemu.conf
-rw-r--r--   1 root root   56 okt  8 22:55 sddm.conf
-rw-r--r--   1 root root  505 okt 26 16:06 systemd.conf
-rw-r--r--   1 root root  398 okt 26 16:06 systemd-remote.conf
-rw-r--r--   1 root root   11 okt 19 14:28 util-linux.conf
-rw-r--r--   1 root root   18 okt 18 23:44 virtualbox.conf
[bobbybabes@ladyluck 4.4.12 ~]# 
```
This is the content of `/usr/lib/sysusers.d/sddm.conf` :
`u sddm - "Simple Desktop Display Manager" /var/lib/sddm`

**Comment by NuLogicSystems** [comment]
Yes, but it doesn't seem to be working properly.
PS. I updated my previous comment..

**Comment by h4nn35** [comment]
This commit https://github.com/sddm/sddm/commit/ba9ab1ddf2c6ebef1fbc95fca75f196d5cfce598 fixes this isse for my setup. 
It's worth a shot.

**Comment by BobbyBabes** [comment]
@h4nn35 This wouldn’t have made it into the Arch Linux package yet : `Last Updated:	2017-10-11 11:18 UTC`. I can’t test until then.

**Comment by NuLogicSystems** [comment]
@BobbyBabes Of course not, Arch pulls releases, not development code. That commit was made only 16 hours ago, is the 130 commit since 0.16.0 was released and submitted by h4nn35 himself.
@h4nn35 Talk about tooting your own horn.

**Comment by BobbyBabes** [comment]
@NuLogicSystems He wrote :
> It's worth a shot.

**Comment by NuLogicSystems** [comment]
@BobbyBabes That why the winking smiley face.
It seems two command were needed for one person over at Manjaro:
sudo systemctl set-default graphical.target
sudo systemctl restart sddm
https://forum.manjaro.org/t/is-sddm-still-screwed/34662/18

**Comment by h4nn35** [comment]
Sorry, that I wasn't clear enough. 
What I meant is that you could try to compile sddm with this patch and see if it solves your problem as it did for me.  == It's worth a shot. :-)

**Comment by sanosis** [comment]
If sddm works after `systemctl restard sddm` you can add a delay to sddm:

1. Define your favorite editor if you haven't already (replace vim with your choice):
`export EDITOR=vim`
2. Edit unit
`systemctl edit sddm`
3. Add to the editor (10s delay):
```
[Service]
ExecStartPre=/bin/sleep 10
```
This will create `/etc/systemd/system/sddm.service.d/override.conf`

**Comment by grayk79** [comment]
@sanosis this is a good workaround for now that doesn't mess with the driver or kernel. I just hate seeing that there are available updates and I just couldn't take that anymore :D Btw a shorter delay works too, I myself used 1 second and everything works just as expected. But I still hope the bug will get fixed soon

**Comment by Gibbz** [comment]
Is there any eta on a fix for this?
 It's a bit annoying having to go around fixing this on all my machines.

**Comment by antonio-rojas** [comment]
Arch Linux users please test 0.16.0-3, I've backported the commit from https://github.com/sddm/sddm/issues/905#issuecomment-344165648

**Comment by BobbyBabes** [comment]
@antonio-rojas @h4nn35 @grayk79 Confirming that `sddm 0.16.0-3` resolves this issue. Thanks to you and all others who helped.

Changed line `MODULES="amdgpu"` back to `MODULES=""`. Recreated `/etc/mkinitcpio.conf`. Recreated `/boot/grub/grub.cfg`.
At the first boot the screen went completely black and then showed an almost full screen of colourful artefacts. But the box booted into sddm. On two subsequent boots, the artefacts were not there anymore.
```
Linux ladyluck 4.14.3-1-ARCH #1 SMP PREEMPT Thu Nov 30 18:33:13 UTC 2017 x86_64 GNU/Linux
sddm 0.16.0-3
xorg-server 1.19.5-1
qt5-base 5.9.3-1
mesa 17.2.6-1
xf86-video-amdgpu 1.4.0-1
```

**Comment by oskarkook** [comment]
@antonio-rojas seems to fix it for me, thanks

**Comment by Musikolo** [comment]
@antonio-rojas, yes, it solves the issue for me too.

Thank you!

**Comment by grayk79** [comment]
Multiple users have reported that the issue is solved(me myself can't test it atm). May I close the thread then?

**Comment by angel6700** [comment]
Before closing I also report that sddm 0.16.0-3 fixes the problem. 
Thank you antonio-rojas.
